aea/protocols/generator.py

```diff
@@ -919,216 +970,958 @@ def _message_class_str(self) -> str:
+        )
+        cls_str += indent + "assert (\n"
+        _change_indent(1)
+        cls_str += indent + "self.dialogue_label.dialogue_opponent_addr\n"
+        cls_str += indent + "== final_dialogue_label.dialogue_opponent_addr\n"
+        _change_indent(-1)
+        cls_str += indent + ")\n"
+        cls_str += indent + "assert (\n"
+        _change_indent(1)
+        cls_str += indent + "self.dialogue_label.dialogue_starter_addr\n"
+        cls_str += indent + "== final_dialogue_label.dialogue_starter_addr\n"
+        _change_indent(-1)
+        cls_str += indent + ")\n"
+        cls_str += indent + "self._dialogue_label = final_dialogue_label\n\n"
+        _change_indent(-2)
+
+        # stats class
+        cls_str += indent + "class {}DialogueStats(object):\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        _change_indent(1)
+        cls_str += (
+            indent
+            + '"""Class to handle statistics for {} dialogues."""\n\n'.format(
+                self.protocol_specification.name
+            )
+        )
+        cls_str += indent + "def __init__(self) -> None:\n"
+        _change_indent(1)
+        cls_str += indent + '"""Initialize a StatsManager."""\n'
+        cls_str += indent + "self._self_initiated = {\n"
+        _change_indent(1)
+        cls_str += indent + "{}Dialogue.EndState.SUCCESSFUL: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_CFP: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_PROPOSE: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_ACCEPT: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        _change_indent(-1)
+        cls_str += indent + "}}  # type: Dict[{}Dialogue.EndState, int]\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "self._other_initiated = {\n"
+        _change_indent(1)
+        cls_str += indent + "{}Dialogue.EndState.SUCCESSFUL: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_CFP: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_PROPOSE: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        cls_str += indent + "{}Dialogue.EndState.DECLINED_ACCEPT: 0,\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        _change_indent(-1)
+        cls_str += indent + "}}  # type: Dict[{}Dialogue.EndState, int]\n\n".format(
+            self.protocol_specification_in_camel_case
+        )
```

Comment:
See above, these end states need to be configurable/depend on performatives

Reply:
Done